src/parser.rs

```diff
@@ -55,65 +55,103 @@ fn parse_expression(tokens: &[Token], options: &LiquidOptions) -> Result<Box<Ren
 /// used internally, from a list of Tokens. This is mostly useful
 /// for correctly parsing complex expressions with filters.
 pub fn parse_output(tokens: &[Token]) -> Result<Output> {
-    let entry = match tokens[0] {
-        Identifier(ref x) => Argument::Var(Variable::new(x)),
-        ref x => Argument::Val(try!(Value::from_token(x))),
+    let mut iter = tokens.iter().peekable();
+    let token = iter.next();
+
+    let entry = match token {
+        Some(&Identifier(ref x)) => Argument::Var(Variable::new(x)),
+        Some(ref x) => Argument::Val(Value::from_token(x)?),
+        _ => return Error::parser("a token", None),
     };
 
+    if let None = iter.peek() {
+        return Ok(Output::new(entry, vec![]));
+    }
+
+    expect(&mut iter, &Pipe)?;
+
     let mut filters = vec![];
+
+    for filter_tokens in tokens.split(|token| token == &Pipe).skip(1) {
+        filters.push(try!(parse_filter(filter_tokens)));
+    }
```

Comment:
I had a lot of difficulty with this because at first I thought named arguments had to be at the end of the call (since the filter gets them as the 3rd arg). However, further investigation revealed that named args can be followed by positional args, which meant that this refactor wasn't needed. 

Splitting the filter chain for parsing means it was easier for me to grok, however it produces less-than-ideal error messages when an unexpected token is present at the end of the filter's token slice, as shown in [this test](https://github.com/cobalt-org/liquid-rust/pull/116/files#diff-2c09afcdc3c420ab0678ba9b5e83959cR363)